staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -272,8 +272,16 @@ func (o *DebugOptions) Validate(cmd *cobra.Command) error {
 	}
 
 	// TargetContainer
-	if len(o.TargetContainer) > 0 && len(o.CopyTo) > 0 {
-		return fmt.Errorf("--target is incompatible with --copy-to. Use --share-processes instead.")
+	if len(o.TargetContainer) > 0 {
+		if len(o.CopyTo) > 0 {
+			return fmt.Errorf("--target is incompatible with --copy-to. Use --share-processes instead.")
+		}
+		if !o.Quiet {
+			// If the runtime doesn't support container namespace targeting this will fail silently, which has caused
+			// some confusion (ex: https://issues.k8s.io/98362), so print a warning. This can be removed when
+			// EphemeralContainers are generally available.
+			fmt.Fprintf(o.Out, "Targeting container %q. Some container runtimes do not support this feature.\n", o.TargetContainer)
```

Comment:
Should this add a "WARNING: " on the beginning, the same way as https://github.com/kubernetes/kubernetes/pull/98250/files does? 

Otherwise, lgtm

Reply:
SGTM. I've added this warning, thanks.